R/flatten.R

```diff
@@ -156,21 +156,21 @@ DropMultipleStatisticsFromTable <- function(
     ## select first statistic from last dimension of table
     ## array() is used to ensure that any dimensions of the
     ## table that are length-1 are not dropped
-    if (n.dims == 2)
-        out <- table[, 1]
-    else if (n.dims == 3)
-        out <- array(table[, , 1], dim = dim(table)[-n.dims])
-    else if (n.dims == 4)
-        out <- array(table[, , , 1], dim = dim(table)[-n.dims])
-    else if (n.dims == 5)
-        out <- array(table[, , , , 1], dim = dim(table)[-n.dims])
-    if (n.dims == 2)
-        names(out) <- dimnames(table)[[1]]
-    else
-        dimnames(out) <- dimnames(table)[-n.dims]
+    n.dim <- getDimensionLength(table)
+    args <- c(list(table), rep(alist(, )[1L], n.dim))
```

Comment:
This is very clever and I have seen you do it before but I wish there was a nice name for this trick